### PR TITLE
Remove Beta from DurableExecutor

### LINKED ISSUE #12083: Remove Beta annotation from DurableExecutor classes
This was Beta in June 2016. Agreed to remove the beta annotation.
https://github.com/hazelcast/hazelcast/pull/12020#discussion_r158004605

### PR DESCRIPTION
**Pull request by pveentjer**
fix #12083

### CODE REVIEW

File: hazelcast/dataset/META-INF/CompilerHints [hunk 1]
```diff
@@ -0,0 +1,5 @@
+dontinline,*.*_all_jmhStub
```
Comment (mmedenjak) [inline hazelcast/dataset/META-INF/CompilerHints]: What's this then? :)

Comment (pveentjer) [reply hazelcast/dataset/META-INF/CompilerHints]: Ah good one :D

This is left over from the code generated by project-x. In my branch the dataset directory has been excluded; but in this branch it isn't.

**Comment by mmedenjak** [comment]
You missed `DurableExecutorConfig`

**Comment by pveentjer** [comment]
doh!

**Pull request merged**